x-pack/heartbeat/monitors/browser/synthexec/synthexec.go

```diff
@@ -213,9 +215,12 @@ func runCmd(
 		return nil, err
 	}
 
-	// Kill the process if the context ends
+	// Get timeout from parent ctx
+	timeout, _ := ctx.Value(SynthexecTimeout).(time.Duration)
+	ctx, cancel := context.WithTimeout(ctx, timeout)
 	go func() {
 		<-ctx.Done()
+
```

Comment:
As we discussed the kill/warning is a problem since this will happen even during runs that are not broken. I think the simplest way to do that would be to define an atomic bool that flips when `cmd.Wait` returns and check that here.

Reply:
Rather than use a switch, which introduces a (small) state dependency between the two routines, I'd rather check the exit status with `if !cmd.ProcessState.Exited() { // kill and log error}`. wdyt?

Reply:
I didn't realize that existed, that's perfect